charts/argo-cd/templates/argocd-server/gateway-httproute.yaml

```
@@ -0,0 +1,48 @@
+{{- if .Values.server.gatewayAPI.enabled }}
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: {{ include "argo-cd.server.fullname" . }}
+  namespace: {{ .Release.Namespace | quote }}
+  labels:
+    {{- include "argo-cd.labels" (dict "context" . "component" .Values.server.name "name" .Values.server.name) | nindent 4 }}
+    {{- with .Values.server.gatewayAPI.labels }}
+      {{- toYaml . | nindent 4 }}
+    {{- end }}
+  {{- with .Values.server.gatewayAPI.annotations }}
+  annotations:
+    {{- range $key, $value := . }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+  {{- end }}
+spec:
+  parentRefs:
+    - group: gateway.networking.k8s.io
+      kind: Gateway
+      name: {{ .Values.server.gatewayAPI.gateway }}
+      {{- with .Values.server.gatewayAPI.gatewaySection }}
```

Could you please add support for referencing Gateways in other namespaces
```suggestion
      {{- with .Values.server.gatewayAPI.gatewayNamespace }}
      namespace: {{ . }}
      {{- end }}
      {{- with .Values.server.gatewayAPI.gatewaySection }}
```